### PR TITLE
working with iframes

### PR DESCRIPTION
**Pull request by ana-os-mo**

### CODE REVIEW

**Review by JuanDelahozMiranda — APPROVED** [review]

**Review by holgiosalos — CHANGES_REQUESTED** [review]
changes are required

File: cypress/e2e/iframe.cy.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    iframePage = new IframePage;
+  });
+
+  it("should able to interact with the iframe elements",
+      {"retries": 2}, () => {
+        const htmlTitle: string = "HTML Tutorial";
+
+        iframePage.visitIframePage();
+
+        iframePage.getFrameTitle(htmlTitle);
```
Comment (holgiosalos) [inline cypress/e2e/iframe.cy.ts]: first test should include these lines

File: cypress/e2e/iframe.cy.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+        iframePage.goToCssPageInFrame();
```
Comment (holgiosalos) [inline cypress/e2e/iframe.cy.ts:22]: this should be part of a second test, where you visit the CSS page and perform the validation that the title is "CSS Tutorial"

File: cypress/page/iframe.page.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    this.navigationBar = "div.w3-bar.w3-left";
+    this.iframeTitle = "#main > h1";
+  }
+
+  public visitIframePage() {
+    cy.visit(this.iframePageURL);
+    cy.frameLoaded(this.iframe);
+  }
+
+  public getFrameTitle(title: string) {
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: if the method is called get, then you should retrieve the title and perform the assertion in the test class

File: cypress/page/iframe.page.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  constructor() {
+    this.iframePageURL = "https://www.w3schools.com/html/html_iframe.asp";
+    this.iframe = "#main iframe[src='default.asp']";
+    this.navigationBar = "div.w3-bar.w3-left";
+    this.iframeTitle = "#main > h1";
+  }
+
+  public visitIframePage() {
+    cy.visit(this.iframePageURL);
+    cy.frameLoaded(this.iframe);
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: probably this is not required in this method

File: cypress/page/iframe.page.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+class IframePage {
+  private iframePageURL: string;
+  private iframe: string;
+  private navigationBar: string;
+  private iframeTitle: string;
+
+  constructor() {
+    this.iframePageURL = "https://www.w3schools.com/html/html_iframe.asp";
+    this.iframe = "#main iframe[src='default.asp']";
+    this.navigationBar = "div.w3-bar.w3-left";
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: why not change the selector to use an attribute, that represents the CSS Tutorial of the navigation bar.

File: cypress/page/iframe.page.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    cy.iframe(this.iframe).find(this.iframeTitle)
+        .should("have.text", title);
+  }
+
+  public goToCssPageInFrame() {
+    const cssBtn = "CSS";
+
+    cy.enter(this.iframe).then((frame) => {
+      frame().find(this.navigationBar).contains(cssBtn).click();
+    });
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts]: why not:
```js
cy.iframe(this.iframe).find(this.navigationBar).click();
cy.frameLoaded(this.iframe, {url: cssPageUrl});
```

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**